---
layout: single
classes: wide
permalink: /workshops/
title: "Workshops"
---

Materials from various workshops I have organized and contributed to. Workshop
materials are normally sufficiently detailed to serve as stand-alone tutorials.
In general the most recent workshop will have the most up-to-date information,
but each workshop is slightly different, and the older materials should still
contain some useful information.

RADSeq Assembly and Analysis
----------------------------
[RADCamp workshops](https://radcamp.github.io/) are designed to introduce
[ipyrad](https://ipyrad.readthedocs.io), a unified and self-contained RADSeq
assembly and analysis framework, which emphasizes simplicity, performance, and
reproducibility. In these workshops we proceed through all the steps necessary
to assemble a typical RADSeq dataset. Additionally, we introduce the ipyrad
‘analysis’ API which provides a powerful, simple, and reproducible interface to
several widely used methods for inferring phylogenetic relationships,
population structure, and admixture.
* [RADCamp Phoenix](https://radcamp.github.io/Phoenix2024/)  (2024; 1 Day; Arizona State University)
* [RADCamp San Francisco](https://radcamp.github.io/SanFrancisco2024/) (2024; 2 Day; California Academy of Sciences)
* [RADCamp Chicago](https://radcamp.github.io/Chicago2024/) (2024; 2 Day; Field Museum of Natural History)
* [RADCamp Chicago](https://radcamp.github.io/Chicago2023/) (2023; 3 Day; Field Museum of Natural History)
* [RADCamp Kigali](https://radcamp.github.io/Kigali2023/) (2023; 2 Day; University of Rwanda)
* [RADCamp NYC](https://radcamp.github.io/NYC2023/) (2023; 4 Day; Columbia University)
* [RADCamp Marseille](https://radcamp.github.io/Marseille2020/) (2020; 3 Day; Online)
* [RADCamp NYC](https://radcamp.github.io/NYC2020/) (2020; ½ Day; Online)
* [RADCamp Lisbon](https://radcamp.github.io/Lisbon2020/) (2020; 1 Day; University of Lisbon)
* [RADCamp NYC](https://radcamp.github.io/NYC2019/) (2019; 4 Day; Columbia University)
* [RADCamp Yale](https://radcamp.github.io/Yale2019/) (2019; ½ Day; Yale University)
* [RADCamp IBS](https://radcamp.github.io/IBS2019/) (2019; Full Day; International Biogeography Society meeting, Malaga, Spain)
* [RADCamp NYC](https://radcamp.github.io/NYC2018/) (2018; 3 Day; Columbia University)
* [RADCamp AF-BIOTA](https://radcamp.github.io/AF-Biota/) (2018; 3 Day; University of São Paulo, Brazil)

Model-based inference in Phylogeography from single species to communities
--------------------------------------------------------------------------
[CompPhylo workshops](https://compphylo.github.io/) are designed to introduce a
myriad of tools for making statistical inference about historical processes
using genetic/genomic data. After introducing statistical approaches in
model-based inferences, participants are be introduced to inference frameworks
that span taxonomic scales from single-species demographic inference, to
multi-species comparative analysis, to inference at the scale of the whole
community. Participants also get hands on experience using approximate
Bayesian computation, supervised machine learning and composite likelihood
methods for model comparison.


* [CompPhylo Mexico City](https://compphylo.github.io/MexicoCity2023/) (2023; ½ Day; SSB Standalone Meeting; UNAM, Mexico City, Mexico)
 * Phylogeographic Temporal Analysis (PTA): Model based comparative phylogeography
with machine learning
* [CompPhylo Oslo](https://compphylo.github.io/Oslo2019/) (2019; 5 Day; University of Oslo)

Multidimensional Biodiversity Data Analysis & Process-Based Modelling
---------------------------------------------------------------------
This is a two-part workshop aimed at rapidly bringing participants to a high
level of proficiency in the management and analysis of multidimensional
biodiversity data. Part I serves as an introduction to data management and
visualization, and serves as a crash course in the data methods that will be
necessary to participate in Part II, which focuses on analyzing biodiversity
data using process-based models and machine learning. 

**Multidimensional biodiversity data: management and analysis**  
Biodiversity researchers must work with an array of data types, including
community composition and abundance information, trait, phylogenetic and genetic
data. Traditionally, studies in ecology and evolution have worked with only one
or a few of these data types. To successfully advance the study of biodiversity
across different levels of organization, biodiversity scientists are nonetheless
finding the need to integrate multiple of these disparate data types into the
same analytical workflow. This workshop will promote learning in the use of
multidimensional data streams, gathering biodiversity scientists and students
from different sub-disciplines to help facilitate integrative,
cross-specialization research.

**Goals of the Workshop:** This two-day workshop will introduce different data
types used by biodiversity scientists in an integrative framework. We will cover
common approaches for working with abundance, trait, phylogenetic, and genetic
data separately, and proceed to methods for working with multiple dimensions of
biodiversity data simultaneously. Finally, we will explore motivations and
platforms for archiving, sharing, and accessing multidimensional biodiversity
datasets as part of the wider scientific community (e.g. GEOME).

**Process-based modeling and statistical inference**  
Process-based models are a powerful framework for generating theoretical
expectations and exploring hypothetical scenarios that can then be linked
directly to multidimensional biodiversity data streams (e.g. joint data on
genetic and species diversity) for hypothesis testing and statistical inference.
While several excellent simulation modeling platforms have been developed in
recent years, these models can be technically sophisticated and difficult to
work with for new users. This workshop will provide an introduction to a
process-modeling approach and hands-on experience working through a complete
workflow, using a user-friendly process model implemented in R: the Rules of
Life Engine (RoLE) model. RoLE is a process-based eco-evolutionary simulation
model which incorporates all relevant biodiversity processes (drift, migration,
selection, speciation) and makes joint predictions of multiple dimensions of
biodiversity data.

**Goals of the Workshop:** This two-day workshop will provide an introduction
to process based modeling using the RoLE model. We will cover the philosophy
and motivation behind process modeling, the use of simulation models to explore
hypotheses and develop theoretical intuition, and statistical methods for using
empirical data to test the hypotheses generated through process modeling.

* [Multidimensional Biodiversity Data](https://role-model.github.io/multidim-biodiv-data/) (2023; 2 Day; Albuquerque, NM)
* [Process-based Modeling and Machine Learning](https://role-model.github.io/process-models-workshop/) (2023; 2 Day; Albuquerque, NM) 
